Faraday_Proxy_Tools/FaradayIO/deviceconfig.py

```diff
@@ -163,13 +175,70 @@ def update_gps(self, gps_boot_bitmask, latitude_str, latitude_dir_str, longitude
 
         :return: Nothing
         """
+        # Setup check variables
+        lat_check = True
+        lat_dir_check = True
+        lon_check = True
+        lon_dir_check = True
+        alt_check = True
+        alt_units_check = True
+
+        # Check Latitude formatting
+
         lat_check = len(latitude_str) <= self.MAX_GPS_LATITUDE_LEN
         lat_dir_check = len(latitude_dir_str) <= self.MAX_GPS_LATITUDE_DIR_LEN
```

Comment:
You never check for only valid inputs which "N" or "S". Nothing else. I put "C" and "X" into mine just fine.

![image](https://cloud.githubusercontent.com/assets/331540/23091212/914c1da8-f565-11e6-8621-44f4c1e64760.png)